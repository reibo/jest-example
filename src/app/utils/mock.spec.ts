describe('mocks', () => {
  it('mock functions', () => {
    function map<T>(items: T[], callback: (x: T) => T) {
      return items.map(item => callback(item));
    }

    const mockCallback = jest.fn(x => x * 2);
    const newList = map([1, 2, 3], mockCallback);

    // The mock function is called once
    expect(mockCallback.mock.calls.length).toBe(3);
    expect(mockCallback).toHaveBeenCalledTimes(3);

    // The first argument of the first call to the function was 0
    expect(mockCallback.mock.calls[0][0]).toBe(1);
    expect(mockCallback).nthCalledWith(1, 1);
    expect(mockCallback).nthCalledWith(2, 2);
    expect(mockCallback).nthCalledWith(3, 3);


    // The return value of the first call to the function was 0
    expect(mockCallback.mock.results[0].value).toBe(2);
    expect(mockCallback).nthReturnedWith(1, 2);
    expect(mockCallback).nthReturnedWith(2, 4);
    expect(mockCallback).nthReturnedWith(3, 6);

    expect(newList).toMatchSnapshot();
  });

  it('Mock return values', () => {
    const vikingMock = jest.fn();
    console.log(vikingMock());
    // > undefined

    vikingMock
      .mockReturnValueOnce('ng-vikings')
      .mockReturnValueOnce('is')
      .mockReturnValueOnce('so')
      .mockReturnValueOnce('cool')
      .mockReturnValue(true);

    console.log(vikingMock(), vikingMock(), vikingMock(), vikingMock(), vikingMock(), vikingMock());
    // > ng-vikings is so cool true true
  });
});
