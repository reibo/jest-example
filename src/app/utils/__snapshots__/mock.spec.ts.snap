// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mocks mock functions 1`] = `
Array [
  2,
  4,
  6,
]
`;
